GoedeDoelenHelpen/ClientApp/src/app/facebook/facebook.component.ts

```diff
@@ -0,0 +1,70 @@
+import { Component, OnInit, TemplateRef, ContentChild  } from '@angular/core';
+import { WindowWrapper } from '../classes/windowwrapper/windowwrapper';
+import { FBAssignModel } from './FBAssignModel';
+import { FBAuthModel, FBBackendResponse} from './FBAuthModel';
+import { AuthenticationService } from '../authentication.service';
+import { Injectable } from '@angular/core';
+import { FbService } from './fb.service';
+import { MatDialog } from '@angular/material';
+
+
+@Component({
+  selector: 'app-facebook',
+  templateUrl: './facebook.component.html',
+  styleUrls: ['./facebook.component.css']
+})
+export class FacebookComponent implements OnInit {
+
+  templateRef: TemplateRef<any>;
+
+  constructor(private windowWrapper: WindowWrapper,
+    private authService: AuthenticationService,
+    private fbService: FbService,
+    private dialog: MatDialog) { }
+
+  ngOnInit() {
+  }
+
+  ngAfterContentInit() {
+    let found: boolean = false;
+    for (var i = 0; i < 10; i++) {
+      if (found == false) {
+        if (this.windowWrapper.nativeWindow.FB != undefined) {
```

Comment:
Weer dezelfde constructie. Is het niet mogelijk om iets dergelijks in een service te stoppen en beschikbaar te stellen als Observable?